flows/actions/send_msg.go

```diff
@@ -156,25 +157,45 @@ func (a *SendMsgAction) getTemplateMsg(run flows.Run, urn urns.URN, channelRef *
 	}
 
 	// next we cross reference with params defined in the template translation to get types
-	params := make(map[string][]flows.TemplateParam, len(translation.Components()))
+	components := make([]flows.TemplateComponent, 0)
+	oldParams := make(map[string][]flows.TemplateParam, len(translation.Components()))
 
-	for key, comp := range translation.Components() {
+	translationComponents := translation.Components()
+
+	buttonIndex := -1
+	for _, comp := range translationComponents {
 		compParams := comp.Params()
+		compType := comp.Type()
+
+		key := compType
+		if strings.HasPrefix(compType, "button/") {
+			buttonIndex += 1
+			key = fmt.Sprintf("button.%d", buttonIndex)
+		}
```

Comment:
I think we're still missing something... name on components https://github.com/nyaruka/rapidpro/pull/5117

So now editor take the list of actual components, and return a list of uuid+name+params components.. and here we can use `name` to match them.

Reply:
The alternative is that template syncing has maintain UUIDs for existing components.. but that's complicated

Reply:
component UUIDs are managed by the flow editor, I do not think the syncing is touching any of that

Reply:
yeah what I meant is we could have RP set them but it would have to be more complex syncing that doesn't just generate new UUIDs every time.